contracts/src/Verifier.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.17;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import "./IVerifier.sol";
+
+/**
+ * @title Verifier contract
+ * @author Seiya Kobayashi
+ */
+// TODO: enable solhint
+/* solhint-disable */
+contract Verifier is IVerifier, Ownable {
```

Comment:
Inheriting `Ownable` for contract owner to be able to fine-tune challenge-generating-algorithm as needed.